contracts/ChainlinkPriceOracle.sol

```diff
@@ -28,37 +54,154 @@ contract ChainlinkPriceOracle is PriceOracle, OracleErrorReporter {
      */
     function getUnderlyingPrice(CToken cToken) public view returns (uint) {
         // Check that a price feed exists for the cToken
-        AggregatorInterface feed = priceFeeds[address(cToken)];
-        require(address(feed) != address(0), "Price feed doesn't exist");
+        AggregatorDetails memory feedDetails = priceFeeds[address(cToken)];
+        require(address(feedDetails.feed) != address(0), "Price feed doesn't exist");
 
         // Get the price
-        int price = feed.latestAnswer();
+        int price = feedDetails.feed.latestAnswer();
         require(price >= 0, "Price cannot be negative");
-        return uint(price);
+
+        return uint(price).mul(uint(10)**feedDetails.decimals);
```

Comment:
`uint(10)**feedDetails.decimals` could overflow as `uint8 decimals` can be a number like 255.

Consider limiting `decimals` to 32 for example. That is more than enough precision and will make this safe.

Reply:
What about limiting to 18 in the setter, since that's the number of decimals required by Compound anyway?

Reply:
I'm in favor of adding a `require(decimals <= 18)` check in `_setPriceFeed`